### PR TITLE
Setup linting with flake8 and plugins

### PR DESCRIPTION
**Pull request by jpd-defra**
This pull request adds the relevant configuration to run flake8 linting as part of a nox-driven pipeline. 

Ideally it could do with testing as part of the CI pipeline, to check that the configuration is correct. 

It also includes a markdown file describing the flake8 module, why and how to use it.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[flake8]
+select = ANN,B,B9,BLK,C,D,DAR,E,F,I,S,W
+ignore = E203,E501,W503
+max-line-length = 80
+max-complexity = 10
+application-import-names = esa_geo_utils,tests
```
Comment (EFT-Defra) [inline .flake8]: This should be our package

```suggestion
application-import-names = sds_data_model,tests
```

File: markdown/linting.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Your `settings.json` file can be configured to select/ignore certain `flake8` error codes. For example the following can be added:
+
+``` json
+"python.linting.flake8Args": ["--ignore=E303"]
+```
+By default, the `settings.json` file will have "Linting in general" and "Linting on file save" set as true.
+
+### As part of a CI pipeline
+
+This project uses GitHub Actions for Continuous Integration. The CI workflow is defined in `.github/workflows/python-package.yml`. This workflow will run the  `black` and `isort` sessions as part of the `format` job, whenever code is pushed to the `develop` branch or pull requests into `develop` are triggered.
```
Comment (EFT-Defra) [inline markdown/linting.md]:
```suggestion
This project uses GitHub Actions for Continuous Integration. The CI workflow is defined in `.github/workflows/python-package.yml`. This workflow will run the  `lint` job, whenever code is pushed to the `develop` branch or pull requests into `develop` are triggered.
```

**Review by EFT-Defra — CHANGES_REQUESTED** [review]
A couple of very major corrections

**Comment by jpd-defra** [comment]
Thanks @EFT-Defra, those two suggestions have been accepted. I won't mess about with resolving those conflicts for now...

**Review by EFT-Defra — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**